cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/ResponseUtils.java

```diff
@@ -112,6 +121,60 @@ public static void writeErrorResponse(HttpServletResponse response,
       responseMessage = errorMessage == null ? "" : errorMessage;
     }
     // Send the CORS Task ID header as part of this error response if 2-step verification is enabled.
-    writeResponseToOutputStream(response, responseCode, json, responseMessage, config);
+    writeResponseToOutputStream(response, responseCode, json, wantJsonSchema, responseMessage, config);
+  }
+
+  private static String getJsonSchema(String responseMessage) {
+    JsonElement response = new JsonParser().parse(responseMessage);
+    return convertNodeToStringSchemaNode(response, null);
+  }
+
+  private static String convertNodeToStringSchemaNode(
+          JsonElement node, String key) {
+    StringBuilder result = new StringBuilder();
+
+    if (key != null) {
+      result.append("\"" + key + "\": { \"type\": \"");
+    } else {
+      result.append("{ \"type\": \"");
+    }
+    if (node.isJsonArray()) {
```

Comment:
From the JSON schema's spec on [array](https://json-schema.org/understanding-json-schema/reference/array.html),looks like we should not iterate all the element, extract and print their schema info, instead we should just do it once?

Also if it is an empty array, can we just output `{ "type": "array" }` instead of `{ "type": "array", "items": []}`?

Reply:
> From the JSON schema's spec on array, looks like we should not iterate all the elements, extract and print their schema info one by one, instead we should just do it once?

The reason why we iterate through every element is that each element can potentially have a different schema. I have modified the code to first put them into a set to avoid any duplicates. However, this will not be perfect as we could still run into cases where some elements in the array have optional fields and some don't, in which case, there will be two entries in items since we do not have enough context to tell they are the same items. This is the limitation of dynamically generating the schema as we do not have the complete schema information at generation time.

> Also if it is an empty array, can we just output { "type": "array" } instead of { "type": "array", "items": []}?
I have added this change. Empty arrays will just output `{ "type": "array" }` now.

Reply:
The issue with print all the element is that the output will be very huge in some case(consider we are print the partition load for 1m partitions in a cluster). 

For optional fields, whether it exists in element or not should purely depend on `verbose`/`super_verbose` parameter in the request, which means in one specific response, it should still be consistent between elements. 

If elements in array are not consistent, it should be a bug in code since it means to be an `array` instead of `tuple`. Therefore for header schema, one element's schema should be enough.

Reply:
Sounds good. I have updated the PR to use the first element for non-empty arrays.